reporter/src/main/kotlin/reporters/freemarker/FreemarkerTemplateProcessor.kt

```diff
@@ -179,6 +183,20 @@ class FreemarkerTemplateProcessor(
         return outputFiles
     }
 
+    /**
+     * Extend the given template [dataModel] by wrapper beans for enum classes, so that these enums can be referenced
+     * directly by templates.
+     * See https://freemarker.apache.org/docs/pgui_misc_beanwrapper.html#jdk_15_enums.
+     */
+    private fun makeEnumsAvailable(dataModel: Map<String, Any>): Map<String, Any> {
+        val beansWrapper = BeansWrapperBuilder(Configuration.VERSION_2_3_30).build()
+        val enumModels = beansWrapper.enumModels
+
+        val enumBeans = ENUM_CLASSES.associate { it.simpleName to enumModels.get(it.name) }
+
+        return dataModel + enumBeans
```

Comment:
Is there a design reason why this already returns the union of `dataModel + enumBeans`? I think I'd prefer this to only return `enumBeans` (that variable could then be inlined) and write at the call site `val dataModel = mapOf(...) + enumModels()` (while renaming this function to `enumModels()` and making it take no arguments).

Reply:
Actually, I'd expect the hard-coded `ENUM_CLASSES` to be a parameter of this function instead, so it gets called like `enumModels(ENUM_CLASSES)`.